elixirkit/lib/elixirkit/server.ex

```diff
@@ -0,0 +1,30 @@
+defmodule ElixirKit.Server do
+  @moduledoc false
+
+  use GenServer
+
+  def start_link(arg) do
+    GenServer.start_link(__MODULE__, arg, name: __MODULE__)
+  end
+
+  @impl true
+  def init(pid) do
+    port = System.fetch_env!("ELIXIRKIT_PORT") |> String.to_integer()
+    {:ok, socket} = :gen_tcp.connect('localhost', port, mode: :binary, packet: :line)
```

Comment:
It is a bit unconventional that the native C#/Swift client is a tcp server and the Elixir server is a tcp client. It has two major benefits though:

1. The native client starts the Elixir server and they need to communicate somehow, e.g. agree on a tcp port. With this approach the native client can bind to _any_ port and just tell the server which one. Otherwise, we'd have to either hardcode the port or find an available port in the client.

2. The Elixir server connects to a tcp server that already is up. Similarly, the native client can wait for the Elixir server to come up by just waiting for the convention, a tcp accept. Otherwise, the native client would need to wait for the tcp server to come up using some kind of busy loop.

Reply:
Yeah, I think this approach is completely fine, safer, and more portable. If there is ever a performance concern, we can provide in-memory transfer for selected platforms.